CIP Cardano Multichain Router b02eb0e3a2bf408a85bdbe6a4ce5603b.md

```diff
@@ -0,0 +1,395 @@
+
+## Router
+
+Cardano will be integrated with Multichain router V3, which was lauched on June 4th 2021. The Router can allow assets to be transferred among all others connected chains including Cardano. Check more about router here:
+
+[https://docs.multichain.org/getting-started/how-it-works/cross-chain-router](https://docs.multichain.org/getting-started/how-it-works/cross-chain-router)
+
+# User Cases
+
+Tokens on different chains can be bridged to Cardano, but let’s simplify it to the scenario between Ethereum and Cardano.
+
+On Cardano side, the Multichan SMPC nodes manage a MPC address, which locks the Cardano native tokens bridged to Ethereum and mint the wrapped tokens bridged from Ethereum.
+
+On Ethereum side, the Multichain SMPC nodes will call Multichain router smart contract to send the Ethereum native token in the pool to users and mint the wrapped tokens bridged from Cardano. 
+
+## A. ****Bridging Ethereum native tokens from Ethereum to Cardano****
+
+1, User send USDC to the USDC pool on Ethereum and indicate the receiving address on Cardano.
+
+2, The SMPC nodes verify the user transaction on Ethereum independently.
+
+3, After the verification, the SMPC nodes work together to sign a minting transaction on Cardano.
+
+4, The SMPC nodes mint wrapped USDC to user’s address on Cardano. 
+
+## B. Bridging wrapped Ethereum native token from Cardano to Ethereum
+
+1, User send wrapped USDC to the SMPC address on Cardano and indicate the receiving address on Ethereum.
+
+2, The SMPC nodes verify the user transacton on Cardano independently.
+
+3, After the verification, the SMPC nodes work together to sign a transfer transaction on Ethereum.
+
+4, The SMPC nodes send the native USDC to user’s address on Ethereum. 
+
+## C. Bridging Cardano native token from Cardano to Ethereum
+
+1, User send native ADA to the SMPC address on Cardano and indicate the receiving address on Ethereum.
+
+2, The  SMPC nodes verify the user transaction on Cardano independently.
+
+3, After the verification, the SMPC nodes work together to sign a minting transaction on Ethereum.
+
+4, The SMPC nodes mint wrapped ADA to user’s address on Ethereum.
+
+## D. Bridging wrapped Cardano native token from Ethereum to Cardano
+
+1, User send wrapped ADA to the wrapped ADA pool on Ethereum and indicate the receiving address on Cardano.
+
+2, The SMPC nodes verify the user transaction on Ethereum independently.
+
+3, After the verification, the SMPC nodes work together to sign a transfer transaction on Cardano.
+
+4, The SMPC nodes send the native ADA to user’s address on Cardano.
+
+# Specification
+
+## Major Modules
+
+### SwapOut transaction
+
+Cardano: Transfer to SMPC address with metadata indicating key/value:
+
+```json
+{
+    "metadata": {
+        "key": "123",
+        "value": {
+            "bind": "target address",
+            "toChainId": 1
+        }
```

Comment:
How would the bridge know which token the user wants to bridge over? The user can include any number of tokens in the UTxO + ADA. The bridge would need to decide which one to take into consideration.